File: src/main/java/ch/uzh/ifi/access/model/SubmissionFile.java
package ch.uzh.ifi.access.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Getter
@Setter
@Entity
@NoArgsConstructor
public class SubmissionFile {
    @Id
    @GeneratedValue
    public Long id;

    @Column(columnDefinition = "text")
    public String content;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "task_file_id")
    public TaskFile taskFile;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "submission_id")
    public Submission submission;

    public Long getTaskFileId() {
        return taskFile.getId();
    }

}

File: src/main/kotlin/ch/uzh/ifi/access/model/SubmissionFile.kt
package ch.uzh.ifi.access.model

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*
import lombok.NoArgsConstructor

@Entity
@NoArgsConstructor
class SubmissionFile {
    @Id
    @GeneratedValue
    var id: Long? = null

    @Column(nullable = false, columnDefinition = "text")
    var content: String? = null

    @JsonIgnore
    @ManyToOne
    @JoinColumn(nullable = false, name = "problem_file_id")
    var problemFile: ProblemFile? = null

    @JsonIgnore
    @ManyToOne
    @JoinColumn(nullable = false, name = "submission_id")
    var submission: Submission? = null
    val problemFileId: Long?
        get() = problemFile!!.id
}